docs/techspec-and-packages.md

```diff
@@ -1 +1,26 @@
-tba
+
+### UI
+The view layer is built with the [React](https://reactjs.org) v.16.6.3 library which enables the composition of encapsulated components to build complex and dynamic layouts that update according to the application state.
+For more information please read the [official docs](https://www.reactjs.org).
+---
+### State Management
+The application state is managed with [Mobx](https://mobx.js.org/index.html) v.5.8.0, a state management solution that aims for ease of use and simplicity to aid development speed and developer experience all while providing app performance and reliability.
+The core idea behind it is to make producing an inconsistent state impossible, by making sure everything that can be derived from the application state will be derived automatically.
+
+- It operates on four major concepts:
+
+1. Make state trackable for Mobx using the **[@observable](https://mobx.js.org/refguide/observable.html)** decorator
+2. to create functions that can automatically derive their value from the state use the **[@computed](https://mobx.js.org/refguide/computed-decorator.html)** decorator
+3. use **[reactions](https://mobx.js.org/refguide/autorun.html)** ( ex. mobx.autorun() ) which are essentially task runners governed by state changes, to produce side effects like logging, making api calls etc.
+4. use **[mobx actions](https://mobx.js.org/refguide/action.html)** only and always on functions that modify state, to allow Mobx to track their invocations
+**[Mobx-react](https://github.com/mobxjs/mobx-react)** provides aditional functionality to bridge React and Mobx.
+---
+### Testing
+
+This project uses [Jest](https://jestjs.io/) as its testing environment, alogng with [Enzyme](https://github.com/airbnb/enzyme) which is a utility library to render react components into the DOM and query the DOM tree for assertions provided by [jest-enzyme](https://github.com/FormidableLabs/enzyme-matchers/tree/master/packages/jest-enzyme#readme).
+- Please note that test coverage must be at 100% to **be able to commit or submit a PR**.
```

Comment:
Please remove this paragraph as this is only an introduction to the technology used and not a code recipe for the project. This note is present in the main readme of the repo.